tests/unit/io/test_eventletreactor.py

```
@@ -12,21 +12,172 @@
+
+        assert conn._check_hostname
+
+    def test_validate_hostname_rejects_mismatch(self):
+        conn = EventletConnection.__new__(EventletConnection)
+        conn.uses_legacy_ssl_options = False
+        conn.endpoint = DefaultEndPoint('1.2.3.4')
+        conn.ssl_options = {}
+        conn._socket = Mock()
+        conn._socket.get_peer_certificate.return_value = _make_certificate('wrong.host')
+
+        with self.assertRaises(Exception):
```

Ruff (0.15.21)

[warning] Do not assert blind exception: `Exception`

(B017)